test/faktory_worker/worker_test.exs

```diff
@@ -449,6 +449,39 @@ defmodule FaktoryWorker.WorkerTest do
       assert result.worker_state == :ok
     end
 
+    test "should log error, sleep, and schedule next fetch worker cannot connect" do
+      expect(FaktoryWorker.SocketMock, :connect, fn _host, _port, _opts ->
+        nil
```

Comment:
For both of these `:connect` mocks can we please return one of the known error types so that it is aligned with the `Socket` behaviour.

I have used `{:error, :econnrefused}` in some of the other tests.

Reply:
I'm returning `nil` because that's what the case block is matching on - https://github.com/SeatedInc/faktory_worker/blob/master/lib/faktory_worker/connection_manager.ex#L67

I think if I wanted to have an explicit error it would need to mock a different module. What do you think?

Reply:
Ah I see. I think we have uncovered some code that could do with refactoring.

The connection `connect` function will always return `{:ok, Connection.t()}` or `{:error, term()}` which in effect is what's being mocked in the test. The connection manager then (for a reason I can't recall) handles that on this line https://github.com/SeatedInc/faktory_worker/blob/master/lib/faktory_worker/connection_manager.ex#L85.

This line should be matching on the error case since it's the only other expected outcome and this would then cause your test to fail.